crates/libs/metadata/src/reader/file.rs

```diff
@@ -333,13 +333,21 @@ impl File {
         }
     }
 
+    /// Returns the string from the `#Strings` stream as referenced by
+    /// the (table, row, column) triple.
+    ///
+    /// # Panics
+    ///
+    /// * When any element of the (table, row, column) triple is invalid.
+    /// * When the offset in the string table is out of bounds.
+    /// * When no null terminator can be found in the string table.
+    /// * When the null-terminated string is not valid utf-8.
     pub fn str(&self, row: usize, table: usize, column: usize) -> &str {
         let offset = self.strings + self.usize(row, table, column);
 
-        unsafe {
-            let len = strlen(self.bytes.as_ptr().add(offset));
-            std::str::from_utf8_unchecked(&self.bytes[offset..offset + len])
-        }
+        let bytes = &self.bytes[offset..];
+        let nul_pos = memchr::memchr(0, bytes).unwrap();
+        std::str::from_utf8(&bytes[..nul_pos]).unwrap()
```

Comment:
Perhaps we should use `expect` instead of `unwrap` so the panic message is more clear.

Reply:
Wouldn't

```rust
let nul_pos = bytes.iter().position(|&c| c == 0).unwrap();
```

accomplish the same without a dependency on an external crate?

Reply:
👍🏻 removed the `memchr` dependency and added explicit `expect` messages